test/tests/problems/reference_residual_problem/tests

```diff
@@ -0,0 +1,26 @@
+[Tests]
+  issues = '#9151'
+  design = 'ReferenceResidualProblem.md'
+  [base]
+    type = 'Exodiff'
+    input = 'reference_residual.i'
+    exodiff = 'reference_residual_out.e'
+    requirement = 'ReferenceResidualProblem shall base convergence on comparison of individual variables with reference quantities'
```

Comment:
```suggestion
    requirement = 'The system shall have the ability to to base base convergence on the comparison of individual variables with with respect to reference quantities of those variables.'
```

Reply:
Isn't the number one requirement for a requirement that it be unambiguous? "The system" seems so ambiguous! What system? I thought my language was very clear and concise and described what we require `ReferenceResidualProblem` to do, e.g. what I am testing for it to do

Reply:
> Isn't the number one requirement for a requirement that it be unambiguous?

Yes, However in traditional development, requirements are written first, sometimes by project managers or other people with less software development experience. They lay down a set of rules or requirements that a new software "system" should satisfy. Arguable, the problem with a long set of requirements is that there is no context to them so there's a lot of repetition in setting them up. It's not typical to say - ok we are working down inside this aspect of the system so let's create a sub block of requirements that roll up to just this section. I've been on projects with hundreds of hand-written requirements that software engineers are supposed to take and run with. It's difficult.

However, from an auditing point of view. The lack of context makes it really easy to single out a requirement, read it gaining all the context from one place and then trace through to the design (where object names and other detail is introduced, not before) and on through to a test that verifies that the requirement is being satisfied.

So at the highest level you can't really do much better than the software "system". Implementation is not supposed to be considered at that level because the software engineer should have the freedom to choose how to implement without over specification. It feels awkward to us, because in practice we usually think about implementation first to achieve a generic goal (perhaps not a formal requirement). Then as we implement we see a lot of corner cases which we implement around creating additional requirements. 

Andrew and I specifically asked about naming objects or even modules (i.e. "The TensorMechanicsModule shall") and we were told not to do it that way because that's getting into implementation details. We were told straight from our NQA-1 consultant and the INL SQA person backed that opinion. A lot of our requirements will need to be overhauled and that will take a lot of time, but I'm trying to prevent new requirements from going in that will need to be revisited.

Hopefully all that background helps. I might need to formalize what I just wrote and add it to the SQA documentation page...
 


 individuals , then software engineers take those